lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -115,7 +116,7 @@ def eval_block
 
         def verify_message
           return true if @expected_message.nil?
-          values_match?(@expected_message, @actual_error.message)
+          values_match?(@expected_message, @actual_error.message.to_s)
```

Comment:
Still seems like the wrong place, as it should already be a string before it gets there, if it was an object that wasn't an exception (and thus has a to_s of it's message) it wouldn't fix it.